src/librustc_trans/mir/block.rs

```diff
@@ -321,7 +321,7 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
 
                 // Get the location information.
                 let loc = bcx.sess().codemap().lookup_char_pos(span.lo);
-                let filename = token::intern_and_get_ident(&loc.file.name);
+                let filename = Symbol::intern(&loc.file.name).as_str();
```

Comment:
Might make sense to change `loc.file.name` to `Symbol` - in fact, any `String` in the compiler that is not on some transient and/or error path, may benefit from it.